mlir/utils/jenkins/Jenkinsfile.nightly-perf

```diff
@@ -0,0 +1,87 @@
+pipeline {
+    agent {
+        docker {
+            image 'rocm/mlir:rocm4.1-latest'
+            args '--user "$(id -u):$(id -g)" --device=/dev/kfd --device=/dev/dri --group-add video -u 0'
+            label 'mlir'
+            alwaysPull true
+        }
+    }
+    stages {
+        stage('Environment') {
+            steps {
+                sh 'cat /etc/os-release'
+                sh '/opt/rocm/bin/rocm-smi'
+            }
+        }
+        stage('Build MIOpen') {
+            steps {
+                checkout scm
+                sh '''
+                    cd ~
```

Comment:
Please don't use ~ here. Instead, use a subdirectory of the workspace (see my MIOpen builds). Using ~ runs the risk of having parallel builds overriding each other, and causes the MIOpen build to not be cleaned away properly.

Reply:
Will everything be cleaned away when the docker container is removed?

Reply:
It might be, but that depends on how the docker container's being set up, if someone decided to be clever with volumes, etc.

Reply:
In fact, I intentionally created MIOpen outside of the workspace (after discussed with Jerry) and relied on the docker container to clean up. The workspace is the whole MLIR project. It is weird to build MIOpen inside the MLIR project. On the other hand, I agree we should keep everything under the workspace as much as possible. I have to say neither method is perfect.

Reply:
I'd say it's not inherently weird to have a subdirectory for the subsidiary thing that we sometimes need in our CI. I'd argue for being in the workspace - if nothing else, that's how all my stuff is set up (so it can use Jenkins's constructs like `dir {}` (that reducing how much random shell we need).